### ISSUE TITLE
FSharp.Core.Unittests do not pass on Mono/OSX

### ISSUE THREAD

**Issue by dsyme** [dotnet/fsharp#4275]
Running 

    ./build.sh net40 test

on OSX after #4273 gave me a hang on FSHarp.Core.Unittests ``Async caching should work``.  It looks like it must be related to Async.Sleep.  Note this is with a freshly compiled FSharp.Core.dll

This is odd, because we run these unit tests as part of CI on OSX in http://github.com/fsharp/fsharp.  We should dig into what is different.

![image](https://user-images.githubusercontent.com/7204669/35467173-2543dda6-0303-11e8-9904-ce1e03fa0425.png)

**Comment by cartermp** [comment]
Closing as this is no longer relevant
